include/api/CDataFrameAnalysisState.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#ifndef INCLUDED_ml_api_CDataFrameAnalysisState_h
+#define INCLUDED_ml_api_CDataFrameAnalysisState_h
+
+#include <core/CProgramCounters.h>
+#include <core/CRapidJsonConcurrentLineWriter.h>
+
+#include <maths/CDataFrameAnalysisStateInterface.h>
+
+#include <api/CDataFrameAnalysisState.h>
+#include <api/ImportExport.h>
+
+#include <cstdint>
+
+namespace ml {
+namespace api {
+
+//! \brief Responsible for collecting data frame analysis job statistics, i.e. memory usage,
+//! progress, parameters, quality of results. The class also implements the functionality to
+//! write the state at different iteration into the results pipe.
+class API_EXPORT CDataFrameAnalysisState : public maths::CDataFrameAnalysisStateInterface {
```

Comment:
I think I'd call this something different. For me "state" would be more suggestive of it somehow being the state needed to perform the analysis. How about `CDataFrameAnalysisStatus` or `CDataFrameAnalysisRunStatistics`?

Reply:
Rename to `Instrumentation`